packages/backend/src/services/RenameService/rename.ts

```diff
@@ -0,0 +1,666 @@
+import {
+    AnyType,
+    assertUnreachable,
+    ChartConfig,
+    ChartType,
+    CustomVis,
+    DashboardDAO,
+    DashboardFilterRule,
+    DashboardFilters,
+    FilterGroup,
+    FilterGroupItem,
+    Filters,
+    isAndFilterGroup,
+    isCustomBinDimension,
+    isFilterRule,
+    isOrFilterGroup,
+    MetricQuery,
+    NameChanges,
+    RenameType,
+    SavedChartDAO,
+    SchedulerAndTargets,
+} from '@lightdash/common';
+
+/* There are different methods to replace model names
+ replaceId: Replace only the start of a string, with the new model prefix
+ replaceReference: Replace the reference string part (eg: ${payment.) with the new model name, needed for SQL strings and more complex references
+ replaceString: Do a full string replacement with the new model name, used only if others don't apply, used in custom viz (vega-lite)
+ and some other functions to deal with undefined, lists and objects
+*/
+export const createRenameFactory = (
```

Comment:
This is a collection of methods to rename ids/references or full strings, you can take a look at tests for more details 

This depends if is a table (prefix) or a full metric/dimension name